### PR TITLE
Feature/hounslow ch290 create service eligibility taxonomies

### PR DESCRIPTION
**Pull request by fred-aghq**
### Summary

https://app.clubhouse.io/ayup-digital-ltd/story/290/create-service-eligibility-taxonomies

Adds a migration that will add Service Eligibility as a top-level Taxonomy item, ready for importing the associated Service Eligibility child taxonomies.

The full CSV for the child taxonomies can be found here (requires Clubhouse access): https://media.clubhouse.io/api/attachments/files/clubhouse-assets/606c8274-5319-4fe9-a210-3ea7c712b38e/60a3d4c6-ddf2-4d27-9943-3664ba11ff1e/service_eligibility_taxonomies.csv

### Development checklist

- [ ] Changes have been made to the API?
  - [ ] If so, the OpenAPI specification has been updated
- [x] Database migrations have been added?
  - [ ] If so, the MySQL Workbench ERD has been updated - N/A - adding data, not changing schema
- [x] The code has been linted `./develop composer fix:style`

### Release checklist

Run migrations.
CSV file is available here: https://media.clubhouse.io/api/attachments/files/clubhouse-assets/606c8274-5319-4fe9-a210-3ea7c712b38e/60a3d4c6-ddf2-4d27-9943-3664ba11ff1e/service_eligibility_taxonomies.csv

### CODE REVIEW

File: database/migrations/2021_05_18_144245_seed_service_eligibility_parent_category.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Support\Facades\Date;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Storage;
+
+class SeedServiceEligibilityParentCategory extends Migration
```
Comment (fred-aghq) [inline database/migrations/2021_05_18_144245_seed_service_eligibility_parent_category.php:8]: I was _tempted_ to make this an (abstract) class from which a migration could be extended, but that felt like over-engineering. depends how much we expect to add new taxonomies in the future.

Comment (appsol) [reply database/migrations/2021_05_18_144245_seed_service_eligibility_parent_category.php:8]: You are correct, that would be over-engineered. This is the 3rd root taxonomy to be added, and the first in some time, so this is not something we expect to happen often.

File: storage/app/service-eligibility/taxonomy.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "id": "AFC14A10-3C3F-4AB1-B23B-0F1D013C8C29",
+    "parent_id": null,
+    "name": "Service Eligibility",
+    "order": 1,
```
Comment (appsol) [inline storage/app/service-eligibility/taxonomy.json:6]: Should be order: 0. Order is 0 indexed.

File: storage/app/service-eligibility/taxonomy.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    "created_at": "2021-05-18 14:00:00",
+    "updated_at": "2021-05-18 14:00:00",
+    "depth": 1
```
Comment (appsol) [inline storage/app/service-eligibility/taxonomy.json:9]: Should be depth: 0 as it does not have a parent ID.

**Review by appsol — CHANGES_REQUESTED** [review]
Alter root category to depth: 0 and order: 0.

File: storage/app/service-eligibility/taxonomy.json (hunk 3 above)
Comment (fred-aghq) [reply storage/app/service-eligibility/taxonomy.json:9]: On it.

**Pull request merged**